File: 2211-k-radius-subarray-averages/k-radius-subarray-averages.java
class Solution {
    public int[] getAverages(int[] nums, int k) {
        
        if(k==0){
            return nums;
        }
        int n=nums.length;
        int windowSize=2*k+1;
        int[] averages=new int[n];
        Arrays.fill(averages,-1);
        long[] prefix=new long[n+1];
        if(windowSize>n){
            return averages;

        }
            for(int i=0;i<n;i++){
                prefix[i+1]=prefix[i]+nums[i];
            }

            for(int i=k;i<(n-k);++i){
                int leftBound=i-k;
                int rightBound=i+k;
                long sum=prefix[rightBound+1]-prefix[leftBound];
                int average=(int)(sum/windowSize);
                averages[i]=average;
            }
            return averages;
        
    }
}

File: 2211-k-radius-subarray-averages/k-radius-subarray-averages.py
class Solution(object):
    def getAverages(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: List[int]
        """
        if(k==0):
            return nums
        n=len(nums);
        windowSize=2*k+1;
        averages=[]
        for i in range(0,n):
            averages.append(-1)
        #averages=[-1]*n
        prefix=[]
        if(windowSize>n):
            return averages
        prefix=[0] * (n + 1)
        for i in range(0,n):
            prefix[i+1]=prefix[i]+nums[i]
        for i in range(k,n-k):
            leftBound=i-k
            rightBound=i+k
            sum=prefix[rightBound+1]-prefix[leftBound]
            average=sum/windowSize
            averages[i]=average
        return averages